### PR TITLE
Speed up s2 gap filler

### PR DESCRIPTION
**Pull request by carioca-au**
Changes create a new DAG which will run over 50 workers to speed up the process

### CODE REVIEW

File: dags/sentinel_2/sentinel-2_gap_filler_2.py [hunk 1]
```diff
@@ -0,0 +1,326 @@
+"""
+# Republish STAC SQS messages from a report of missing S3 objects
+
+This DAG is intended to be run manually, and must be passed a configuration JSON object
+specifying `offset`, `limit` line numbers of the report and `s3_report_path` specifying
+the path to the gap report file.
+
+Eg:
+```json
```
Comment (carioca-au) [inline dags/sentinel_2/sentinel-2_gap_filler_2.py:9]: Limit is still valid if we want to put a limit per worker

File: dags/sentinel_2/sentinel-2_gap_filler_2.py [hunk 2]
```diff
@@ -0,0 +1,326 @@
...
+MAX_ACTIVE_RUNS = 30
+
+DEFAULT_ARGS = {
+    "owner": "RODRIGO",
+    "start_date": datetime(2020, 7, 24),
+    "email": ["systems@digitalearthafrica.org"],
+    "email_on_failure": False,
+    "email_on_retry": False,
+    "retries": 0,
+    "limit_of_processes": 50,
```
Comment (carioca-au) [inline dags/sentinel_2/sentinel-2_gap_filler_2.py:51]: Here we have the limit of workers

File: dags/sentinel_2/sentinel-2_gap_filler_2.py [hunk 3]
```diff
@@ -0,0 +1,326 @@
...
+    ][0:3000]  # Todo remove limit when go to PROD
+
+    if len(missing_scene_paths) < 1:
+        raise Exception('No Items found')
+
+    logging.info(f"Number of scenes found {len(missing_scene_paths)}")
+    logging.info(f"Example scenes: {list(missing_scene_paths)[0:10]}")
+
+    # This code will create chunks of 50 scenes and run over the workers
+    max_list_items = math.ceil(len(missing_scene_paths) / DEFAULT_ARGS["limit_of_processes"])
```
Comment (carioca-au) [inline dags/sentinel_2/sentinel-2_gap_filler_2.py:193]: This calculation ensures the number of scenes to be sent per worker

File: dags/sentinel_2/sentinel-2_gap_filler_2.py [hunk 4]
```diff
@@ -0,0 +1,326 @@
...
+
+    # TODO Do not run for lower than 50 items in the file
+    PUBLISH_MESSAGES_FOR_MISSING_SCENES = [
+        PythonOperator(
+            task_id=f"publish_messages_worker_{idx}",
+            python_callable=prepare_and_send_messages,
+            op_args=["{{ dag_run.conf.limit }}", missing_file_chunks[idx]],
+            on_success_callback=task_success_slack_alert,
+        )
+        for idx in range(DEFAULT_ARGS["limit_of_processes"])
```
Comment (carioca-au) [inline dags/sentinel_2/sentinel-2_gap_filler_2.py:323]: Running the limit of workers

File: dags/sentinel_2/sentinel-2_gap_filler_2.py [hunk 5]
```diff
@@ -0,0 +1,326 @@
...
+        doc_md=__doc__,
+) as dag:
+    missing_file_chunks = get_missing_stac_files()
+
+    # TODO Do not run for lower than 50 items in the file
+    PUBLISH_MESSAGES_FOR_MISSING_SCENES = [
+        PythonOperator(
+            task_id=f"publish_messages_worker_{idx}",
+            python_callable=prepare_and_send_messages,
+            op_args=["{{ dag_run.conf.limit }}", missing_file_chunks[idx]],
```
Comment (carioca-au) [inline dags/sentinel_2/sentinel-2_gap_filler_2.py:320]: The TODO above explains the issue here if you run for a list smaller than the limit of workers

**Review by alexgleith — APPROVED** [review]

**Pull request merged**